src/prismo/pl/_plotting.py

```diff
@@ -965,3 +990,44 @@ def factor_sparsity_histogram(model: PRISMO, bins: int = 50, nrow: int | None =
     return _plot_sparse_probabilities_histogram(
         model.get_sparse_factor_probabilities("numpy"), bins=bins, nrow=nrow, ncol=ncol
     )
+
+
+def _prettify_factor_names(
+    factors: pd.Series,
+    replacements: dict[str, str] = None,
+    abbreviation_length: int = 40,
+    end_with_database: bool = True,
+) -> pd.Series:
+    """Prettify factor names.
+
+    Replace underscores with spaces. If `replacements` is not empty, replace the keys with the values.
+    Shorten the labels to `abbreviation_length` characters. If `end_with_database` is `True`,
+    put the database name in square brackets at the end of the label.
+
+    Example:
+    `Reactome ABC` becomes `ABC [R]`, if `replacements` is `{"Reactome": "R"}` and `end_with_database` is `True`.
+
+    Args:
+        factors: The factor names to prettify.
+        replacements: A dictionary of replacements to make.
+        abbreviation_length: The maximum length of the labels.
+        end_with_database: Whether to end the labels with the database name. The database name is
+            assumed to be in square brackets in the beginning of the label.
+    """
+    if not isinstance(abbreviation_length, int) or abbreviation_length <= 0:
+        raise ValueError("`abbreviation_length` must be a positive integer.")
+
+    # Replace whitespace
+    factors = factors.str.replace("_", " ", regex=False)
+    # Replace dict values
+    if replacements is not None:
+        for key, value in replacements.items():
+            factors = factors.str.replace(key, value, regex=False)
+    # Shorten string length and put `...` at the end if abbreviation is needed
+    factors = factors.apply(lambda x: x[:abbreviation_length] + "..." if len(x) > abbreviation_length else x)
+    # Put database name in the end
+    if end_with_database:
+        # Turn `[X] ABC` into `ABC [X]`
+        factors = factors.str.replace(r"(\[.*\]) (.*)", r"\2 \1", regex=True)
```

Comment:
That regex will also match e.g. `test [abc] def`. Add a `^` add the beginning to force it to match the beginning of the string: `r"^(\[.*\]) (.*)"`